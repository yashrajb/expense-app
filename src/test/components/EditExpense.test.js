import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage.js';
import expenses from '../fixture/commonData.js';

test('should render',() => {
    const editExpense = jest.fn();
    const history = { push:jest.fn() };
    const removeExpense = jest.fn();
    const result = shallow(<EditExpensePage editExpense={editExpense} removeExpense={removeExpense} history={history} expense={expenses[2]} />);
    expect(result).toMatchSnapshot();

});

test('should render addexpense',() => {
    const editExpense = jest.fn();
    const history = { push:jest.fn() };
    const removeExpense = jest.fn();
    const result = shallow(<EditExpensePage editExpense={editExpense} removeExpense={removeExpense} history={history} expense={expenses[2]} />);
    result.find("ExpenseForm").prop("onSubmit")(expenses[2]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id,expenses[2]);

});

test('should render removeexpense',() => {
    const editExpense = jest.fn();
    const history = { push:jest.fn() };
    const removeExpense = jest.fn();
    const result = shallow(<EditExpensePage editExpense={editExpense} removeExpense={removeExpense} history={history} expense={expenses[2]} />);
    result.find("button").simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
        id:expenses[2].id
    });
    

});

