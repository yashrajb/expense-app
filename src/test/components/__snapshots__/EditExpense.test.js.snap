// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <ExpenseForm
    expenses={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit card",
        "id": "123456789abcdefghijkl",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
